packages/vitest/src/node/core.ts

```diff
@@ -429,12 +429,17 @@ export class Vitest {
     return testFiles
   }
 
+  private isMatchPattern(id: string, pattern: readonly string[]) {
+    const relativeId = relative(this.config.dir || this.config.root, id)
```

Comment:
Is `this.config.dir` needed here? Isn't exclude/include/includeSource relative to the root?

```suggestion
    const relativeId = relative(this.config.root, id)
```

Also, maybe we want to directly make the path `relative` in isTargetFile, or before calling it (we are already calling slash), instead of creating a private function that is potentially called several times (and results in more complexity).